### DISCUSSION TITLE
Serde Support 2023 December Update

### DISCUSSION THREAD

**Discussion by thomas-k-cameron** [smithy-lang/smithy-rs#3293] [General]
Congratulations on reaching GA!

I wanted to make a write up summarize the current situation on adding serde attributes.

Here is an overview of open PRs.

## PRs for adding tests or refactoring
These PRs are for refactoring or tests; It does not affect functionalities of the crates.
Please take a look at the each PR for further detail.

- https://github.com/smithy-lang/smithy-rs/pull/2850
- https://github.com/smithy-lang/smithy-rs/pull/2851
- https://github.com/smithy-lang/smithy-rs/pull/2803

## Non Trivial PRs
- https://github.com/smithy-lang/smithy-rs/pull/2615
This is the master PR; It had over 30 changed files, and I divided into smaller sub-PRs to improve the reviewing experience.  
This is the PR where it make changes to `*Generator.kt` exists. Once this PR is merged all sdks are going to be shipped with the `serde` attribute.

- https://github.com/smithy-lang/smithy-rs/pull/2650
This one adds `serde` decorator.   
I received a suggestion to add a compiler warning;    
You can find my lengthy comment talking about calling `panic!()` during compilation; This is because I mistakenly believed that by `compiler warning` it meant something that forces the compilation to fail.

- https://github.com/smithy-lang/smithy-rs/pull/2775/
This is the PR for adding compile-time benchmarks.   
CI-script for benchmarking was merged before but it was reverted due to an error.

**Comment by thomas-k-cameron** [comment]
@jdisanti 

I'm going to mention you. I hope my write up will help!  
There are some merge-conflicts; Let me come back to you as soon as I fix them up. 

I will try to ready them during the holidays.

**Reply by jdisanti** [reply]
Thank you for the additional context!

**Reply by thomas-k-cameron** [reply]
Following PRs are passing all the tests. I believe it is ready for review.

- https://github.com/smithy-lang/smithy-rs/pull/2850
- https://github.com/smithy-lang/smithy-rs/pull/2803
- https://github.com/smithy-lang/smithy-rs/pull/2650

Concerning https://github.com/smithy-lang/smithy-rs/pull/2615, I think I will be making changes to it once all the other PRs are merged. (Sub-PR got it's own changes; I believe there are bunch of conflicts.)
